### ISSUE TITLE
PathString over-encodes

### ISSUE THREAD

**Issue by Tratcher** [aspnet/HttpAbstractions#660]
Ported from https://katanaproject.codeplex.com/workitem/462

"The resource path delimiters "(" and ")" are encoded which violates RFC 3986. Please see section 3.3. This breaks a lot of OData clients that rely on a specific URL convention. The offending code can be found in Microsoft.Owin.PathString:ToUriComponent(). 

http://services.odata.org/OData/OData.svc/GetProductsByCategoryId(categoryId=2) 
is encoded as 
http://services.odata.org/OData/OData.svc/GetProductsByCategoryId%28categoryId=2%29 
which is incorrect."

AspNetCore uses a different encoder, but it likely has similar behaviors here.
https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs#L67

```
pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
              / "*" / "+" / "," / ";" / "="
pct-encoded   = "%" HEXDIG HEXDIG
unreserved    = ALPHA / DIGIT / "-" / "." / "_" / "~"
```

**Comment by troydai** [comment]
Here's a prototype:  https://github.com/troydai/HttpAbstractions660/blob/master/PrototypeEncoder.cs
```
Category: unreserved
     Input: /abc123.-_~
PathString: /abc123.-_~
 Prototype: /abc123.-_~
              
  Category: colon
     Input: /:
PathString: /%3A
 Prototype: /:
              
  Category: at
     Input: /@
PathString: /@
 Prototype: /@
              
  Category: sub-delims
     Input: /!$&'()*+,;=
PathString: /!$%26%27()*%2B,;%3D
 Prototype: /!$&'()*+,;=
              
  Category: reserved
     Input: /?#[]
PathString: /%3F%23%5B%5D
 Prototype: /%3F%23%5B%5D
              
  Category: pct-encoding
     Input: /单行道
PathString: /%E5%8D%95%E8%A1%8C%E9%81%93
 Prototype: /%E5%8D%95%E8%A1%8C%E9%81%93
              
  Category: mixed
     Input: /index/单行道=(x*y)
PathString: /index/%E5%8D%95%E8%A1%8C%E9%81%93%3D(x*y)
 Prototype: /index/%E5%8D%95%E8%A1%8C%E9%81%93=(x*y)
```

**Comment by troydai** [comment]
Fixed in 150b4708f158c0cdcc2d49f2ec72f3e82c6c64e4
